### PR TITLE
feat(apm): Add APM for `<OrganizationDetails>` and `<ProjectsDashboard>` [SEN-1057]

### PR DESCRIPTION
**Pull request by billyvg**
Add simple metrics for `<OrganizationDetails>` and
`<ProjectsDashboard>`.

Depends on https://github.com/getsentry/sentry/pull/14961

### CODE REVIEW

**Comment by eurica** [comment]
Cool! So nice to see this.  Are those lines in profiler.jsx something every developer would have to write or just because of something specific we're doing?

File: src/sentry/static/sentry/app/api.tsx [hunk 1]
```diff
@@ -340,7 +347,9 @@ export class Client {
           );
         },
         complete: (jqXHR: JQueryXHR, textStatus: string) => {
-          Sentry.finishSpan(requestSpan);
+          // TODO(kamil): We forgot to add this to Spans interface
+          (requestSpan as any).finish();
```
Comment (kamilogorek) [inline src/sentry/static/sentry/app/api.tsx]: For now we can use:
```js
(requestSpan as Sentry.Span).finish();
```

File: src/sentry/static/sentry/app/utils/apm.jsx [hunk 2]
```diff
@@ -7,45 +7,103 @@ function startTransaction() {
...
+const requests = new Set([]);
+const renders = new Set([]);
 let flushTransactionTimeout = undefined;
-function finishTransaction(delay) {
+let interruptFlush = false;
+
+const hasActiveRequests = () => requests.size > 0;
+const hasActiveRenders = () => renders.size > 0;
+
+export function finishTransaction(delay) {
```
Comment (kamilogorek) [inline src/sentry/static/sentry/app/utils/apm.jsx]: This method should probably also reset `interruptFlush` value

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/utils/apm.jsx [hunk 3]
```diff
@@ -7,45 +7,103 @@ function startTransaction() {
...
+  if (flushTransactionTimeout) {
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/apm.jsx]: Refactored these a bit

File: src/sentry/static/sentry/app/api.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/api.tsx]: Moved to where we start the span

**Comment by billyvg** [comment]
> Cool! So nice to see this. Are those lines in profiler.jsx something every developer would have to write or just because of something specific we're doing?

This is not necessarily a representation of what our SDK will be like for our clients.

**Review by kamilogorek — APPROVED** [review]

**Pull request merged**